pkg/backend/httpstate/backend.go

```diff
@@ -1255,3 +1255,33 @@ func (c httpstateBackendClient) GetStackOutputs(ctx context.Context, name string
 func (c httpstateBackendClient) DownloadPlugin(ctx context.Context, plug workspace.PluginInfo) (io.ReadCloser, error) {
 	return c.backend.DownloadPlugin(ctx, plug, false, display.Options{})
 }
+
+// GetStackResourceOutputs returns the resource outputs of type (if any) for a stack, or an error if
+// the stack cannot be found. Resources are retrieved from the latest stack snapshot, which may
+// include ongoing updates. `typ` optionally specifies the type of resource to retrieve, formatted
+// using the Pulumi URN format (e.g., `kubernetes:core/v1:Service`).
+func (c httpstateBackendClient) GetStackResourceOutputs(
+	ctx context.Context, name, typ string) (resource.PropertyMap, error) {
+	ref, err := c.backend.ParseStackReference(name)
+	if err != nil {
+		return nil, err
+	}
+	s, err := c.backend.GetStack(ctx, ref)
+	if err != nil {
+		return nil, err
+	}
+	if s == nil {
+		return nil, errors.Errorf("unknown stack %q", name)
+	}
+	snap, err := s.Snapshot(ctx)
+	if err != nil {
+		return nil, err
+	}
+	pm := resource.PropertyMap{}
+	for _, r := range snap.Resources {
+		if typ == "" || string(r.Type) == typ {
+			pm[resource.PropertyKey(r.ID)] = resource.NewObjectProperty(r.Outputs)
+		}
+	}
```

Comment:
Do we need two implementations of this?  Can't we just forward to the underlying backend like we do for `GetStackOutputs`?

Reply:
Totally not necessary -- very WIP. We will fix.